src/Core/Components/List/ListComponentBase.razor.cs

```diff
@@ -369,6 +342,26 @@ protected override void OnParametersSet()
 
     }
 
+    /// <inheritdoc />
+    protected override bool TryParseValueFromString(string? value, [MaybeNullWhen(false)] out string? result, [NotNullWhen(false)] out string? validationErrorMessage)
+        => this.TryParseSelectableValueFromString(value, out result, out validationErrorMessage);
+
+    /// <inheritdoc />
+    protected override string? FormatValueAsString(string? value)
+    {
+        // We special-case bool values because BindConverter reserves bool conversion for conditional attributes.
+        if (typeof(TOption) == typeof(bool))
+        {
+            return (bool)(object)value! ? "true" : "false";
```

Comment:
By definition, `value` can be null. Maybe you could add a test like this (to remove the `!`)
```csharp
if (value is not null && typeof(TOption) == typeof(bool))
```